verification/store.go

```diff
@@ -75,3 +69,25 @@ func LoadX509TrustStore(path string) (*X509TrustStore, error) {
 
 	return &trustStore, nil
 }
+
+func validateCerts(certs []*x509.Certificate, path string) error {
+	// to prevent any trust store misconfigurations, ensure there is at least
+	// one certificate from each file.
+	if len(certs) < 1 {
+		return fmt.Errorf("could not parse a certificate from %q, every file in a trust store must have a PEM or DER certificate in it", path)
+	}
+
+	for _, cert := range certs {
+		if !cert.IsCA {
+			if err := cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature); err != nil {
```

Comment:
cert.CheckSignature does not guarantee self-signed right? If the cert is not a CA, then we need to make sure it's the only certificate in this certs slice, and it's self-signed.

Reply:
Updated to use `CheckSignatureFrom` instead.